src/sentry/db/models/fields/slug.py

```diff
@@ -16,16 +16,33 @@ class IdOrSlugLookup(Lookup):
     lookup_name = "id_or_slug"
 
     def as_sql(self, compiler, connection):
-        # We can ignore the lhs because we know that the lhs will
-        # query either 'slug' or 'id'
+        lhs, lhs_params = self.process_lhs(compiler, connection)
         rhs, rhs_params = self.process_rhs(compiler, connection)
 
+        # Use Django's method to properly quote table and column names
+        table_name = lhs.split(".")[0] if "." in lhs else None
```

Comment:
What would the `lhs` look like before splitting. Can you add an example in comments?